utilization_summary.py

```diff
@@ -20,64 +21,69 @@ def all_users_from_file(userfile, args):
+        user_list[x] = users[x] + utilization_calculator(users[x][0], months, time_entries, today)
+    write_output(args, user_list, months)
     print(color.PURPLE+"TOCK BLOCKS:"+color.END+" Completed generating the utilization summary. Please view the report in the file "+ args.outfile +".")
 
-def util_csv(user, months, time_entries):
+def find_months(today, args):
+    print(args)
+    months = [0,0]
+    if args.beginmonth is None:
+        months[0] = today.month - 2
+    if args.lastmonth is None:
+        months[1] = today.month + 1
+    if months[1] == 0:
+        months[1] = month_name_list.index(args.lastmonth)+1
+    if months[0] == 0:
+        months[0] = month_name_list.index(args.beginmonth)+1
+        if months[0] >= months[1]:
+            months[0] = months[0] - 12
+    return months
+
+def utilization_calculator(user, months, time_entries, today):
     # Grab user
     user_entries = tock_blocks.get_user_entries(user, time_entries)
     # Calculate each month billable/ utilization / total for that user
     user_values = [0] * (months[1]-months[0])
     array_ind = 0
     for x in range(months[0], months[1]):
-        mStart = ""
-        mEnd = ""
-        lastDay = "01"
-        if x < 9:
-            mStart = "0"+str(x)
-            mEnd = "0"+str(x+1)
-        elif(x == 9):
-            mStart = "0"+str(x)
-            mEnd = "10"
-        elif(x == 12):
-            mStart = "12"
-            mEnd = "12"
-            lastDay = "31"
-        else:
-            mStart = str(x)
-            mEnd = str(x+1)
-        month_time_entries = tock_blocks.get_entries_in_time_period("2017-"+mStart+"-01", "2017-"+mEnd+"-"+lastDay, user_entries, 'util')
+        mStart = calculateMonthYear(x, today)
+        month_time_entries = tock_blocks.get_entries_in_month(mStart+"-01", user_entries)
         billable_hours = calc_billable_hours(month_time_entries)
         internal_hours = calc_internal_hours(month_time_entries)
         total_hours = calc_total_hours(month_time_entries)
-        billable_perc = 0.0
-        internal_perc = 0.0
+        billable_percent = 0.0
+        internal_percent = 0.0
         if (total_hours > 0):
-            billable_perc = round(billable_hours/total_hours*100, 1)
-            internal_perc = round(internal_hours/total_hours*100, 1)
-        utilizable_perc = round((billable_perc + internal_perc), 1)
-        user_values[array_ind]=[billable_perc,internal_perc, utilizable_perc]
-        array_ind += 1
+            billable_percent = round(billable_hours/total_hours*100, 1)
+            internal_percent = round(internal_hours/total_hours*100, 1)
+        utilizable_percent = round((billable_percent + internal_percent), 1)
+        user_values[array_ind]=[billable_percent,internal_percent, utilizable_percent]
+        array_ind +=1
     return user_values
 
+def calculateMonthYear(monthValue, today):
+    year_to_use = today.year
+    x = monthValue
+    if monthValue <= 0:
```

Comment:
This will work if the year is off by one, which is probably good enough for most situations.

Reply:
Ah, I see that the documentation specifies up to 12 months. If you get ambitious and want to expand that, I think it could be done.

Reply:
Yeah, I know, but for now. I think that is a future feature.